core-tests/src/test/java/org/mule/runtime/core/DefaultMuleMessageSerializationTestCase.java

```diff
@@ -97,8 +96,8 @@ String getContent()
         public NonSerializableToByteArray()
         {
             super();
-            registerSourceType(new SimpleDataType<NonSerializable>(NonSerializable.class));
-            setReturnDataType(DataTypeFactory.BYTE_ARRAY);
+            registerSourceType(DataType.builder(NonSerializable.class).build());
```

Comment:
For this usage would it not be ideal to have something like `DataType.for(NonSerializable.class)` rather than forcing use of builder?